### PR TITLE
6503 pg heron front end vunerabilities

### PR DESCRIPTION
**Pull request by siddharthsatyakam**
This script connect e-compliance and figure outs study team. 

When I changed from 2.7 to 4.3, new groovy did not allow to access java.lang.object was not accessible. (Dan was accessing protective parameter ( java.lang.object ) which is not allowed in the latest version)

To fix this 
1. removed '@immutable' which does not allow chiild class and also defines API classes automatically like get and put.
2. Define explicate constructor for class ``, it takes argument from cli and define args
3. groovy provided explicit params for ssl (SSL is turned off since last 10 years and also here.)

MAde chanegs to allow the constructors (explicit) to assitgn the CLI args to the variables and also modify connections string for ecompliance as per new groovy 4.3

### CODE REVIEW

File: src/main/groovy/kumc_bmi/StudyTeam.groovy [hunk 1]
```diff
@@ -18,6 +18,7 @@ import groovy.json.JsonOutput
 import groovy.sql.Sql
 import groovy.transform.CompileStatic
 import groovy.transform.Immutable
+import groovy.util.Eval.*
```
Comment (lav-patel) [inline src/main/groovy/kumc_bmi/StudyTeam.groovy]: can you import only what you need ?

File: src/main/groovy/kumc_bmi/StudyTeam.groovy [hunk 2]
```diff
@@ -90,20 +92,29 @@ class StudyTeam {
...
+      /*raw_args = args1.toArray(new String[args1.size()]);*/
+      ArrayList<String> raw_args = (ArrayList<String>)args1
+      String[] str_eq_raw_args = raw_args.toArray(new String[raw_args.size()]);
+      this.logger.warning(str_eq_raw_args.toString())
+      this.args_given = str_eq_raw_args
+    }
 
     boolean flag(String it) {
-        args.contains(it)
+        args_given.contains(it)
```
Comment (lav-patel) [inline src/main/groovy/kumc_bmi/StudyTeam.groovy]: I don't see args as reserved word in https://docs.oracle.com/cloud/latest/big-data-discovery-cloud/BDDDE/rsu_transform_unsupported_features.htm#BDDDE-reference_B4D419B4A7634CDD8E312D3AF68F8277

File: src/main/groovy/kumc_bmi/StudyTeam.groovy [hunk 3]
```diff
@@ -90,20 +92,29 @@ class StudyTeam {
...
     }
-
+
```
Comment (lav-patel) [inline src/main/groovy/kumc_bmi/StudyTeam.groovy]: remove additional space

File: src/main/groovy/kumc_bmi/StudyTeam.groovy [hunk 4]
```diff
@@ -90,20 +92,29 @@ class StudyTeam {
...
+/*@Immutable*/
 @CompileStatic
 class CLI {
-    String[] args
+    private Logger logger = Logger.getLogger("")
+    String[] args_given
+    
+    CLI(args1) {
+      /*raw_args = args1.toArray(new String[args1.size()]);*/
+      ArrayList<String> raw_args = (ArrayList<String>)args1
```
Comment (lav-patel) [inline src/main/groovy/kumc_bmi/StudyTeam.groovy:102]: converting string to array of string

Comment (lav-patel) [reply src/main/groovy/kumc_bmi/StudyTeam.groovy:102]: args1 = "[a,b,c]"
raw_args = Arraylist['a','b','c']
str_eq_raw_args = ['a','b','c']

File: src/main/groovy/kumc_bmi/StudyTeam.groovy [hunk 5]
```diff
@@ -28,7 +29,8 @@ class StudyTeam {
         def fail = { ex -> System.err.println(ex); System.exit(1) }
         def dbAccess = { String u, Properties p -> DriverManager.getConnection(u, p) }
         def listen = { InetSocketAddress a, int b -> HttpServer.create(a, b) }
-        run(new CLI(args: args), System.out, fail, dbAccess, listen)
```
Comment (lav-patel) [inline src/main/groovy/kumc_bmi/StudyTeam.groovy:31]: (args: args) is provided by '@Immutable'

**Review by lav-patel — CHANGES_REQUESTED** [review]

File: src/main/groovy/kumc_bmi/StudyTeam.groovy (hunk 3 above)
Comment (siddharthsatyakam) [reply src/main/groovy/kumc_bmi/StudyTeam.groovy]: Done!!!!

**Comment by siddharthsatyakam** [comment]
Docker working great!!!!!

Frontend good!!!

File: src/main/groovy/kumc_bmi/StudyTeam.groovy (hunk 1 above)
Comment (lav-patel) [reply src/main/groovy/kumc_bmi/StudyTeam.groovy]: SS: removed line 21

**Review by lav-patel — APPROVED** [review]

**Pull request merged**